control/discovery.py

```diff
@@ -339,6 +340,30 @@ def __exit__(self, exc_type, exc_value, traceback):
             self.omap_state.cleanup_omap()
             self.omap_state = None
 
+        if self.selector:
+            with self.lock:
+                try:
+                    for key in self.conn_vals:
+                        self.selector.unregister(self.conn_vals[key].connection)
+                        self.conn_vals[key].connection.close()
+                except Except as ex:
+                    pass
```

Comment:
Looks like try/except should be inside for loop. WDYT?

```suggestion
                for key in self.conn_vals:
                    try:
                        self.selector.unregister(self.conn_vals[key].connection)
                        self.conn_vals[key].connection.close()
                   except Except as ex:
                        pass
```

Reply:
I guess it's better. I just took the existing code, but as I change it anyway I can catch the exception for a single socket and continue with the others in case it fails.